Release alertsmanagement microsoft.alerts management 2022 09 01 preview

anprokup opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

anprokup committed:
Adds base for updating Microsoft.AlertsManagement from version stable/2021-04-01 to version 2022-09-01-preview

anprokup committed:
Updates readme

anprokup committed:
Updates API version in new specs and examples

anprokup committed:
added public network access

anprokup committed:
validation fixes

anprokup commented:
Hi @akning-ms ,
I got the following error while running Autorest validation:

ERROR (OperationsAPIImplementation/R3023/ARMViolation): Operations API must be implemented for '/providers/Microsoft.AlertsManagement/operations'.
    - file:///C:/work/azure-rest-api-specs/specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2022-09-01-preview/SmartDetectorAlertRulesApi.json:36:2 ($.paths)
    
But the operations API is already implemented and is up-to-date in this location: 
specification\alertsmanagement\resource-manager\Microsoft.AlertsManagement\stable\2019-03-01\AlertsManagement.json

can this error be suppressed?

Thanks.

raosuhas commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2022-09-01-preview/SmartDetectorAlertRulesApi.json:
```diff
@@ -0,0 +1,773 @@
+          "items": {
+            "$ref": "#/definitions/DetectorParameterDefinition"
+          },
+		  "x-ms-identifiers": []
```
>		 [](http://example.com/codeflow?start=0&length=2)

nit : use spaces not tabs

anprokup replied:
also, I am not sure what is causing the remaining errors. the errors and their explanations are not clear enough

anprokup committed:
fixed "npm prettier" issues.

akning-ms commented:
> Hi @akning-ms , I got the following error while running Autorest validation:
> 
> ERROR (OperationsAPIImplementation/R3023/ARMViolation): Operations API must be implemented for '/providers/Microsoft.AlertsManagement/operations'. - file:///C:/work/azure-rest-api-specs/specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2022-09-01-preview/SmartDetectorAlertRulesApi.json:36:2 ($.paths)
> 
> But the operations API is already implemented and is up-to-date in this location: specification\alertsmanagement\resource-manager\Microsoft.AlertsManagement\stable\2019-03-01\AlertsManagement.json
> 
> can this error be suppressed?
> 
> Thanks.

Hi @anprokup, The reason is the tag in readme.md dont include the swagger file which include operation API, you can solve it by adding it in the tag. refer to previous tag

anprokup committed:
added tags for other API files

anprokup committed:
added more API files to fix readme.md

anprokup committed:
fixed casing for operation paths

anprokup committed:
Revert "fixed casing for operation paths"

This reverts commit 1404b50ff74df369c93d429f60fd3124f6c6f9fa.

anprokup commented:
> 

Hi, thanks for the assistance.
I added the missing files, but now I get an error for different API methods.
This error is because I fixed the casing for the resource namespace (to be Pascal Case), but this causes a diff between the old APIs. I can't change thie back since the Pascal-Case is a must according to another validation.

akning-ms commented:
You dont have to fix casing issue in linter as it will cause breaking change.
For your PR.  seems you removed some path like cross API breaking change reported. if it is true, you may need get breaking change review board approval
For linter diff, there is 2 excpetions, if you can not fix it, you can ping @jianyexi  for help

anprokup committed:
fixed case of paths

anprokup committed:
fixed lint diff

anprokup committed:
fixing default tag

anprokup commented:
> I fixed the issues that caused the breaking change errors. The only problems remaining are the case issues in the paths/namespaces and the MISSING_APIS_IN_DEFAULT_TAG. I am not sure how to fix the MISSING_APIS_IN_DEFAULT_TAG problem even after looking into other examples.
> 
> can you please assist?

@akning-ms can you please assist with the above issue?

akning-ms commented:
> > I fixed the issues that caused the breaking change errors. The only problems remaining are the case issues in the paths/namespaces and the MISSING_APIS_IN_DEFAULT_TAG. I am not sure how to fix the MISSING_APIS_IN_DEFAULT_TAG problem even after looking into other examples.
> > can you please assist?
> 
> @akning-ms can you please assist with the above issue?

@jianyexi to approve the Linter error.  @tianxchen-ms to confirm  MISSING_APIS_IN_DEFAULT_TAG is valid error
BTW, you should remove package.json & package.lock.json in PR and remove label "DoNotMerge" if need me merge
